contracts/axelar-gateway/src/storage_types.rs

```diff
@@ -24,3 +24,16 @@ pub enum DataKey {
     MessageApproval(MessageApprovalKey),
     RotationExecuted(BytesN<32>),
 }
+
+#[contracttype]
+#[derive(Clone, Debug)]
+pub enum AuthDataKey {
```

Comment:
since the contract storage is the same, using two different data keys might cause collisions due to enum overlap? You can look into it/ask their support

Reply:
You can use the same enum, just add a section comment to distinguish. We can think about interesting ways to abstract this. One option is to define an enum variant
`DataKey::Auth(AuthDataKey)`, although usage is a bit annoying since you always have to wrap it.

Reply:
How the contract storage works in Soroban is quite interesting. So I tested with those data key names within the same contract:

```
#[derive(Clone)]
#[contracttype]
pub enum DataKey {
    Owner,
    Test,
    Ahram
}

#[derive(Clone)]
#[contracttype]
pub enum AuthDataKey {
    Ahram,
    Test,
    Owner
}

#[derive(Clone)]
#[contracttype]
pub enum DataKeyTest {
    Test,
    Ahram,
    Owner
}
```
I was able to test it with an example, I confirmed that the order of variables and name of data key doesn't matter, `the enum variable name in data key` only matters.

What it means is, 
`&DataKey::Ahram`, `&AuthDataKey::Ahram`, and `&DataKeyTest::Ahram` are pointing the same storage slot.
For example, to access storage slot `Ahram`, the code below points to the same storage slot.
```
env.storage().instance().set(&DataKeyTest::Ahram, &test);

env.storage().instance().get(&AuthDataKey::Ahram);
env.storage().instance().get(&DataKeyTest::Ahram);
env.storage().instance().get(&DataKey::Ahram);
```